File: js/Maximum Number of Integers to Choose From a Range I.js
var maxCount = function(banned, n, maxSum) {
    banned.sort((a, b) => a - b);

    let sum = 0, count = 0;

    for (let i = 1; i <= n; i++) {
        if (isBanned(banned, i)) continue;

        if (sum + i > maxSum) break;

        sum += i;
        count++;
    }

    return count;
};
function isBanned(banned, x) {
    let left = 0, right = banned.length - 1;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        if (banned[mid] === x) return true;
        if (banned[mid] < x) left = mid + 1;
        else right = mid - 1;
    }

    return false;
}
// Test cases
console.log(maxCount([1, 6, 5], 5, 6)); // Output: 2
console.log(maxCount([1, 2, 3, 4, 5, 6, 7], 8, 1)); // Output: 0
console.log(maxCount([11], 7, 50)); // Output: 7

File: python/Maximum Number of Integers to Choose From a Range I.py
from bisect import bisect_left

class Solution:
    def maxCount(self, banned, n, maxSum):
        banned.sort()
        total_sum = 0
        count = 0

        for i in range(1, n + 1):
            if self.is_banned(banned, i):
                continue
            
            if total_sum + i > maxSum:
                break
            
            total_sum += i
            count += 1
        
        return count
    
    def is_banned(self, banned, x):
        index = bisect_left(banned, x)
        return index < len(banned) and banned[index] == x

# Test cases
if __name__ == "__main__":
    solution = Solution()
    
    # case 1
    print("Test case 1 output:", solution.maxCount([1, 6, 5], 5, 6)) # Expected output: 2
    
    # case 2
    print("Test case 2 output:", solution.maxCount([1, 2, 3, 4, 5, 6, 7], 8, 1)) # Expected output: 0
    
    # case 3
    print("Test case 3 output:", solution.maxCount([11], 7, 50)) # Expected output: 7 